### DISCUSSION TITLE
KEDA Scaler for Azure DevOps Pipelines - PAT Refresh Question

### DISCUSSION THREAD

**Discussion by takis-kapas** [kedacore/keda#6349] [General]
Actually there are two questions but they are related to the Azure DevOps Pipeline KEDA Scaler.

I have setup a working infrastructure with a Trigger Authentication and ScaledJob. The TA is pulling the Azure DevOps PAT from a Namespace Generic Secret object to authenticate to the ADO Agent Pool. I recycle the ADO PAT every 90 days. When I tried to update the Namespace Secret value, the Trigger Authentication Object was still pulling the "old" secret values. I had to delete and re-created the TA and SJ Objects to re-authenticate successfully with the new Secret value to the ADO Pool. Is this by design?

Would there be a possibility (maybe new feature?) for the Trigger Authentication to authenticate to ADO using a Service Principal secret/key, or even Managed Identity instead of a PAT? Both are supported as users in ADO.

**Comment by zroubalik** [comment]
Yeah, this request sound reasonable, are you willing to open a PR?

**Reply by takis-kapas** [reply]
Hi @zroubalik and thank you for your response. 

I don't have any code written/ready for this Issue so that I am able to open a proposed PR.

I just ran on this problem on a production use case and submitted the issue for consideration of the contributors/maintainers of the KEDA project/repo.
